PlutoHelperAgent/SharedFunctions.m

```diff
@@ -0,0 +1,236 @@
+    sleep(1);
+    
+    NSHTTPCookie *cookie;
+    NSHTTPCookieStorage *cookieJar = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+    for (cookie in [cookieJar cookies]) {
+        NSLog(@"%@", cookie);
+    }
+    
+    NSString *URLToUse2 = [NSString stringWithFormat: @"%@/api/isLoggedIn", [[NSUserDefaults standardUserDefaults] stringForKey:@"project_locker_url"]];
+    
+    NSURL *url2 = [NSURL URLWithString:URLToUse2];
+    NSURLSessionConfiguration *config2 = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session2 = [NSURLSession sessionWithConfiguration:config2];
+    
+    NSMutableURLRequest *request2 = [[NSMutableURLRequest alloc] initWithURL:url2];
+    request2.HTTPMethod = @"GET";
+    [request2 setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    NSDictionary * headers = [NSHTTPCookie requestHeaderFieldsWithCookies:[cookieJar cookies]];
+    [request2 setAllHTTPHeaderFields:headers];
+    NSLog(@"Request %@", request2);
+    NSLog(@"Request %@", request2.allHTTPHeaderFields);
+    
+    NSError *error2 = nil;
+    
+    connectionStatusString = @"Connection Failed";
+    
+    if (!error2) {
+        NSURLSessionDataTask *uploadTask2 = [session2 dataTaskWithRequest:request2 completionHandler:^(NSData *data2,NSURLResponse *response2,NSError *error2) {
+            NSString *datastring2 = [[NSString alloc] initWithData:data2 encoding:NSUTF8StringEncoding];
+            NSLog(@"Data2 %@", datastring2);
+            NSLog(@"Response2 %@", response2);
+            if (error2 != NULL) {
+                NSLog(@"Error2 %@", error2);
+                
+            }
+            
+            NSString *compareThisPartOfTheString = [datastring2 substringToIndex:14];
+            
+            if ([compareThisPartOfTheString isEqual: @"{\"status\":\"ok\""]) {
+                
+                connectionStatusString = @"Connection Okay";
+
+            }
+ 
+        }];
+        
+        [uploadTask2 resume];
+
+    }
+    
+    sleep(1);
+
+    return connectionStatusString;
+    
+}
+
++ (void) logout_of_project_server {
+    
+    NSHTTPCookieStorage *cookieJar = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+    
+    NSString *URLToUse = [NSString stringWithFormat: @"%@/api/logout", [[NSUserDefaults standardUserDefaults] stringForKey:@"project_locker_url"]];
+    
+    NSURL *url = [NSURL URLWithString:URLToUse];
+    NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:config];
+    
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
+    request.HTTPMethod = @"GET";
+    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    NSDictionary * headers = [NSHTTPCookie requestHeaderFieldsWithCookies:[cookieJar cookies]];
+    [request setAllHTTPHeaderFields:headers];
+    
+    NSError *error = nil;
+    
+    if (!error) {
```

Comment:
is there any point in this comparison? seems like `error` is set to nil on the line before so this will always be true

Reply:
No. Sorry. This was left over rubbish from cut and pasted code that I did not notice because it was not breaking anything. Removed.